app/components/password/reset-password.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+const { Component, set, get } = Ember;
+import { task } from 'ember-concurrency';
+
+export default Component.extend({
+  /**
+   * @property password
+   * @default String
+   */
+  password: '',
+  /**
+   * @property passwordConfirmation
+   * @default String
+   */
+  passwordConfirmation: '',
+  /**
+   * @property error
+   */
+  error: null,
+
+  /**
+   * @property resetPasswordTask
+   * @param password
+   * @param passwordConfirmation
+   */
+  resetPasswordTask: task(function* (password, passwordConfirmation) {
+    try {
+      yield get(this, 'resetPassword')(password, passwordConfirmation);
```

Comment:
Yeah to it is passed in as closure action, so just need to get it off the props and call that function.  This way also allows me to patch that action in the integration test.